src/Services/CleanupService.php

```diff
@@ -191,97 +167,48 @@ private function removeInactiveSlaveGroups(): void
             return;
         }
 
-        $runningProcesses = [];
+        $this->setActiveSlaveGroups();
+
 
         foreach ($this->activeProbes as $device => $probes){
 
             if(!isset($this->activeGroups[$device])){
                 continue;
             }
 
-            if(\is_array($probes)){
-                $probes = implode(' ' . $this->path . '/' .$device. '/',  $probes);
-            }
-
-            $probes = $this->path . '/' .$device . '/' . $probes;
+            $items = $this->concatCollection($probes, $this->path . '/' .$device. '/');
 
-            // !!! Max x amount of processes !!!!
-            // Do not modify unless you know what
-            // You're doing
+            $storedSlaves = $this->storage->listItems($items, false);
```

Comment:
Dangerous to use the raw data you receive from an ls command. What if the next storage implementation cannot use ls ?